### DISCUSSION TITLE
Is there a way to use multiple OnlineManagers?

### DISCUSSION THREAD

**Discussion by tienshiao** [TanStack/query#3553] [Q&A]
Since React Query can be used with async tasks beyond just network requests (like maybe Web Bluetooth or Web USB), is there a way to use multiple OnlineManagers to support scenarios where maybe you're only connected to the network but not the peripheral or vice versa where you're only connected to the peripheral but you're offline?

**Comment by TkDodo** [comment]
v4 has a `networkMode` setting that takes care of this problem: https://react-query-beta.tanstack.com/guides/network-mode

**Reply by tma-isbx** [reply]
I saw that discussion (in the issues) though I was thinking more along the lines of treating the peripheral as a separate network such that if the BLE connection was reestablished polling/refetching would resume or if the BLE connection is lost queries would pause independently of the network used by fetch/axios.

It seems like to accomplish something like that with Network Mode, I would tag my non-fetch/axios queries as `always` and then independently track the BLE connection status and add a guard in those async functions. It seems like I'd probably lose the ability to rely on `fetchStatus` in those contexts also.

**Reply by TkDodo** [reply]
I see. For that to work, each query would kinda need to have its own onlineManager 🤔
